daemon/logger/jsonfilelog/read.go

```diff
@@ -305,20 +308,13 @@ func followLogs(f *os.File, logWatcher *logger.LogWatcher, notifyRotate chan int
 		if !since.IsZero() && msg.Timestamp.Before(since) {
 			continue
 		}
+		if !until.IsZero() && msg.Timestamp.After(until) {
+			continue
+		}
 		select {
 		case logWatcher.Msg <- msg:
 		case <-ctx.Done():
 			logWatcher.Msg <- msg
-			for {
```

Comment:
Looks like that was originally added in https://github.com/moby/moby/commit/c0391bf5545afef5e675138556c39e4c0e9bf91b /cc @cpuguy83 you're probably better to tell :smile: